src/app/utils/field-validation/validators/mobileNoValidator.ts

```diff
@@ -0,0 +1,48 @@
+import { chain, left, right } from 'fp-ts/lib/Either'
+import { pipe } from 'fp-ts/lib/function'
+
+import { IMobileField } from 'src/types/field'
+import { ResponseValidator } from 'src/types/field/utils/validation'
+import { ISingleAnswerResponse } from 'src/types/response'
+
+import {
+  isMobilePhoneNumber,
+  startsWithSgPrefix,
+} from '../../../../shared/util/phone-num-validation'
+
+import { notEmptySingleAnswerResponse } from './common'
+
+type MobileNoValidator = ResponseValidator<ISingleAnswerResponse>
+type MobileNoValidatorConstructor = (
+  mobileNumberField: IMobileField,
+) => MobileNoValidator
+
+const mobilePhoneNumberValidator: MobileNoValidatorConstructor = () => (
+  response,
+) => {
+  return isMobilePhoneNumber(response.answer)
+    ? right(response)
+    : left(`MobileNoValidator:\t answer is not a valid mobile phone number`)
+}
+
+const prefixValidator: MobileNoValidatorConstructor = (homeNumberField) => (
+  response,
+) => {
+  if (
```

Comment:
nit: return ternary for easier parsing? and reverse the order so we don't need to read so many negations

```ts
return (mobileNoField.allowIntlNumbers || startsWithSgPrefix(response.answer))
  ? right(response)
  : left(`MobileNoValidator:\t answer is not an SG number but intl numbers are not allowed`)
```

[de morgan's law](https://csawesome.runestone.academy/runestone/books/published/csawesome/Unit3-If-Statements/topic-3-6-DeMorgan.html) btw:
current: `!a && !b`
to reverse order:
`!(!a && !b) === !!a || !!b === a || b`

Reply:
cool stuff, thanks for pointing out!